walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -63,6 +79,37 @@
       freeTimes.add(TimeRange.fromStartEnd(startTime, TimeRange.END_OF_DAY, /* inclusive */ true));
     }
 
-    return freeTimes;
+    // Repeat the exact same search, but this time find meetings
+    // that only mandatory attendees can attend.
+    startTime = TimeRange.START_OF_DAY;
+    ArrayList<TimeRange> freeTimesMandatory = new ArrayList<>();
+
+    // If there are restrictions for only mandatory attendees.
+    if (busyTimesMandatory.size() > 0) {
```

Comment:
Don't need this if statement.

Reply:
Needed this to pass the optionalAttendeesWithoutGaps test (if there are only optional attendees, but none of them can attend any meetings, the code should return an empty list). Without this statement, the code returns the entire day because it assumes you have mandatory attendees who are free the whole day.